samples/posix/syslog/src/main.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#define N_PRIOS 8
+
+int main(void)
+{
+	int prios[N_PRIOS] = {
+		LOG_EMERG,   LOG_ALERT,  LOG_CRIT, LOG_ERR,
+		LOG_WARNING, LOG_NOTICE, LOG_INFO, LOG_DEBUG,
+	};
+
+	openlog("main", LOG_PID | LOG_CONS | LOG_NOWAIT, LOG_LOCAL7);
+	(void)setlogmask(LOG_MASK(-1));
+
+	for (size_t i = 0; i < N_PRIOS; ++i) {
+		syslog(i, "syslog priority %d", prios[i]);
+	}
+
+	closelog();
+
+	/* yield briefly to logging thread */
+	sleep(100);
```

Comment:
1 second should be enough for the logging thread to finish its work

```suggestion
	sleep(1);
```